Capri.Web/Controllers/ProposalController.cs

```diff
@@ -0,0 +1,122 @@
+
+        public ProposalController(
+            UserManager<User> userManager,
+            IProposalCreator proposalCreator,
+            IProposalDeleter proposalDeleter,
+            IProposalGetter proposalGetter,
+            IProposalUpdater proposalUpdater)
+        {
+            _userManager = userManager;
+            _proposalCreator = proposalCreator;
+            _proposalDeleter = proposalDeleter;
+            _proposalGetter = proposalGetter;
+            _proposalUpdater = proposalUpdater; 
+        }
+
+        [HttpGet("{id:Guid}")]
+        public async Task<IActionResult> Get(Guid id)
+        {
+            var result = await _proposalGetter.Get(id);
+            if (result.Successful())
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return BadRequest(result);
+            }
+        }
+
+        [HttpGet]
+        public IActionResult GetAll()
+        {
+            var results = _proposalGetter.GetAll();
+            if (results.Successful())
+            {
+                return Ok(results);
+            }
+            else
+            {
+                return BadRequest(results);
+            }
+        }
+
+        [Authorize(Roles = "promoter")]
+        [HttpPost]
+        public async Task<IActionResult> Create([FromBody] ProposalRegistration registration)
+        {
+            var user = getCurrentUser();
+
+            var result = await _proposalCreator.Create(registration, user.Result);
+            if (result.Successful())
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return BadRequest(result);
+            }
+        }
+
+        [Authorize(Roles = "promoter")]
+        [HttpPut("{id:Guid}")]
+        public async Task<IActionResult> Update(Guid id, [FromBody] ProposalUpdate update)
+        {
+            var user = getCurrentUser();
+
+            var result = await _proposalUpdater.Update(id, update, user.Result);
+            if (result.Successful())
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return BadRequest(result);
+            }
+        }
+
+        [Authorize(Roles = "promoter")]
+        [HttpDelete("{id:Guid}")]
+        public async Task<IActionResult> Delete(Guid id)
+        {
+            var result = await _proposalDeleter.Delete(id);
+            if (result.Successful())
+            {
+                return Ok(result);
+            }
+            else
+            {
+                return BadRequest(result);
+            }
+        }
+
+        private Task<User> getCurrentUser()
+        {
+            //var sub = HttpContext?.User.Claims.FirstOrDefault(c => c.Type == System.Security.Claims.ClaimTypes.NameIdentifier).Value;
+            var userId = User.FindFirst(ClaimTypes.NameIdentifier)?.Value;
+            return _userManager.FindByIdAsync(userId);
```

Comment:
Actually 
`await _userManager.GetUserAsync(HttpContext.User);` will be enough so i don't think you need to wrap it over method